src/TestSuite/IntegrationTestTrait.php

```diff
@@ -655,20 +664,20 @@ protected function _addTokens(string $url, array $data): array
         if ($this->_csrfToken === true) {
             $middleware = new CsrfProtectionMiddleware();
             $token = null;
-            if (!isset($this->_cookie['csrfToken']) && !isset($this->_session['csrfToken'])) {
+            if (!isset($this->_cookie[$this->_csrfCookieName]) && !isset($this->_session[$this->_csrfCookieName])) {
                 $token = $middleware->createToken();
-            } elseif (isset($this->_cookie['csrfToken'])) {
-                $token = $this->_cookie['csrfToken'];
+            } elseif (isset($this->_cookie[$this->_csrfCookieName])) {
+                $token = $this->_cookie[$this->_csrfCookieName];
             } else {
-                $token = $this->_session['csrfToken'];
+                $token = $this->_session[$this->_csrfCookieName];
```

Comment:
Given that we're using this property in the session as well, should it be `_csrfKeyName`?

Reply:
Good point, this variable has been updated with the suggested name.